packages/forms-web-app/src/controllers/examination/file-upload/handleMultipleFileUploadsWithErrors.js

```diff
@@ -0,0 +1,29 @@
+const { multipleFileValidations } = require('./fileValidation');
+const { saveFileToDisk } = require('./fileManagement');
+const { addFileToSession } = require('./fileSessionManagement');
+
+const handleMultipleFileUploadsWithErrors = async (session, fileToUpload) => {
+	const errorSummary = [];
+	let errorMessage = '';
+	for (const document of fileToUpload) {
+		const error = multipleFileValidations(document);
+		if (!error) {
+			const savedFileData = await saveFileToDisk(document);
+			addFileToSession(session, {
+				...savedFileData,
+				raw: document
+			});
+		} else {
+			errorSummary.push({ text: error });
```

Comment:
lift this out and return an array of errors for the summary list and map outside of the function